packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -120,37 +119,36 @@ abstract class MultiChildLayoutDelegate {
   /// Creates a layout delegate.
   ///
   /// The layout will update whenever [relayout] notifies its listeners.
-  MultiChildLayoutDelegate({ Listenable relayout }) : _relayout = relayout;
+  MultiChildLayoutDelegate({ Listenable? relayout }) : _relayout = relayout;
 
-  final Listenable _relayout;
+  final Listenable? _relayout;
 
-  // TODO(ianh): make these late final
-  /*late*/ Map<Object/*!*/, RenderBox>/*!*/ _idToChild;
-  /*late*/ Set<RenderBox/*!*/>/*!*/ _debugChildrenNeedingLayout;
+  Map<Object, RenderBox>? _idToChild;
+  Set<RenderBox>? _debugChildrenNeedingLayout;
```

Comment:
can't these be late final non-nullable?

Reply:
No because it is used to init `debugPreviousChildrenNeedingLayout` and it would fail at the first call.